apps/studio/src/features/editing-experience/components/EditLinkPreview.tsx

```diff
@@ -0,0 +1,80 @@
+import type { IsomerSitemap } from "@opengovsg/isomer-components"
+import { useAtomValue } from "jotai"
+
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { editLinkSchema } from "~/pages/sites/[siteId]/links/[linkId]"
+import { trpc } from "~/utils/trpc"
+import { linkAtom } from "../atoms"
+import PreviewWithoutSitemap from "./PreviewWithoutSitemap"
+import { ViewportContainer } from "./ViewportContainer"
+
+export const EditLinkPreview = (): JSX.Element => {
+  const { summary } = useAtomValue(linkAtom)
+  const { linkId, siteId } = useQueryParse(editLinkSchema)
+  const [permalink] = trpc.page.getFullPermalink.useSuspenseQuery(
+    {
+      pageId: linkId,
+      siteId,
+    },
+    { refetchOnWindowFocus: false },
+  )
+
+  const [{ parent }] = trpc.resource.getParentOf.useSuspenseQuery({
+    resourceId: String(linkId),
+    siteId,
+  })
+
+  const [children] = trpc.collection.getSiblingsOf.useSuspenseQuery({
+    siteId,
+    resourceId: linkId,
+  })
+
+  const parentPermalink = permalink.split("/").slice(0, -1).join("/")
+  const linkSuffix = permalink.split("/").slice(-1).join("/")
+  const parentTitle = parent?.title || "Collection"
+
+  const siteMap = {
+    id: "root",
+    permalink: "/",
+    lastModified: "2024-10-14T07:05:08.803Z",
+    layout: "homepage",
+    title: "An Isomer Site",
+    summary: "",
+    children: [
+      {
+        id: "collection",
+        permalink: parentPermalink,
+        lastModified: "02 May 2023",
+        layout: "collection",
+        title: parentTitle,
+        summary: "Learn more about PREPARE here.",
+        children: children.map((props) => ({
+          ...props,
+          date: "12 May 2023",
```

Comment:
issue(non-blocking): the date should be today's date, otherwise might be confusing to the editor

Reply:
actually this didn't quite work - oddly enough, only may worked. i'm not too sure why because this format `dd MMM yyyy` seems to be valid? lmk if you have any idea what could be the issue